eth-types/src/evm_types/memory.rs

```diff
@@ -417,6 +398,155 @@ impl Memory {
     }
 }
 
+/// Represents a range of memory addresses of [start,end).
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct MemoryRange {
+    start: MemoryAddress,
+    end: MemoryAddress,
+}
+
+impl MemoryRange {
+    /// Creates a new memory range with the provided start and length.
+    pub fn new_with_length(
+        start: impl Into<MemoryAddress>,
+        length: impl Into<MemoryAddress>,
+    ) -> Self {
+        let start = start.into();
+        let end = start + length.into();
+        Self { start, end }
+    }
+
+    /// length of the memory range.
+    #[inline(always)]
+    pub fn length(&self) -> MemoryAddress {
+        self.end - self.start
+    }
+}
+
+/// Represents a 32 bytes .
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub struct MemoryWordRange {
+    /// a memory range aligned to 32 bytes
+    range: MemoryRange,
+    /// length of the original range in bytes
+    original_length: MemoryAddress,
```

Comment:
original_length seems redundant, not use by outside ?

Reply:
trying to make it more generalised by keeping the original information